engine/src/cmd/drawable.cpp

```diff
@@ -262,3 +267,122 @@ bool Drawable::animationRuns() const
 {
     return !done;
 }
+
+
+void Drawable::Split( int level )
+{
+    Unit *unit = static_cast<Unit*>(this);
+
+    if (game_options.split_dead_subunits)
+        for (un_iter su = unit->getSubUnits(); *su; ++su)
+            (*su)->Split( level );
+    Vector PlaneNorm;
+    for (unsigned int i = 0; i < nummesh();) {
+        if (this->meshdata[i]) {
+            if (this->meshdata[i]->getBlendDst() == ONE) {
+                delete this->meshdata[i];
+                this->meshdata.erase( this->meshdata.begin()+i );
+            } else {i++; }} else {this->meshdata.erase( this->meshdata.begin()+i ); }}
```

Comment:
we don't need to shorten the code block; please expand them out for easier comprehension